prow/github/client.go

```diff
@@ -562,7 +562,7 @@ func newClient(
 				Timeout: maxRequestTime,
 				Transport: &oauth2.Transport{
 					Source: newReloadingTokenSource(getToken),
-					Base:   graphQLTransport,
+					Base:   baseRoundTripperWithPreviewHeaders,
```

Comment:
In the v3 client we usually set them on the methods that need them rather than globally: https://github.com/kubernetes/test-infra/blob/2efad71f085991710643e572dbde606377a807b1/prow/github/client.go#L4039

Reply:
Okay, so I assume this means in tide when we call the check method.

I could also add it here globally and only add the header when the calls goes to the check API. This would make it easier to avoid future problems when the checks API is called from another place.

I assume you prefer the first option. WDYT?

Reply:
Yeah, please just add it to the relevant methods

Reply:
@alvaroaleman PTAL :)